addons/l10n_pl/tests/test_move.py

```diff
@@ -0,0 +1,58 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from freezegun import freeze_time
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+from odoo.tests import tagged
+from odoo import fields
+
+
+@tagged('post_install_l10n', 'post_install', '-at_install')
+class TestAccountPL(AccountTestInvoicingCommon):
+
+    @classmethod
+    def setUpClass(cls, chart_template_ref='pl'):
+        super().setUpClass(chart_template_ref=chart_template_ref)
+
+        with freeze_time('2024-07-10'):
+            cls.today = fields.Date.today().strftime('%Y-%m-%d')
+
+        cls.dollar_currency = cls.env['res.currency'].search([('name', '=', 'USD')])
+        cls.PL_currency = cls.env['res.currency'].search([('name', '=', 'PLN')])
+        cls.invoice_a = cls.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': cls.partner_a.id,
+            'date': cls.today,
+            'invoice_date': cls.today,
+            'currency_id': cls.dollar_currency.id,
+            'company_id': cls.env.company.id,
+            'invoice_line_ids': [(0, None, {
+                'name': 'test_product',
+                'product_id': cls.product_a.id,
+                'currency_id': cls.dollar_currency.id,
+                'quantity': 1.0,
+                'price_unit': 1000.0,
+            })]
+        })
+
+    def test_pl_out_invoice_onchange_accounting_date(self):
+        self.invoice_a.delivery_date = '2024-03-31'
+        self.assertEqual(self.invoice_a.invoice_line_ids[0].currency_rate, 1.0)
+
+        self.env['res.currency.rate'].create([
+            {
+                'name': '2024-04-28',
+                'rate': 1.0,
+                'currency_id': self.PL_currency.id,
+                'company_id': self.env.company.id,
+            },
+            {
+                'name': '2024-04-28',
+                'rate': 4.2799058421,
+                'currency_id': self.dollar_currency.id,
+                'company_id': self.env.company.id,
+            },
+        ])
```

Comment:
```suggestion
        self.env['res.currency.rate'].create({
                'name': '2024-04-28',
                'rate': 4.2799058421,
                'currency_id': self.currency_usd.id,
                'company_id': self.invoice_a.company_id.id,
        })
```
we only need the one rate here (since the company uses PLN as a currency, and the rate for PLN to PLN will always be 1.00)

Reply:
Done